docs/index.rst

```diff
@@ -106,23 +106,79 @@ You can use pip and make to quickly set up a development environment::
+        * It is preferable to use `alpha` data type for default string values for easier debug procedure
+        * In case any workaround is introduced, it is necessary to provide corresponding BZ ID directly into the code docstring
+        * All linting and formatting/style checks would be enforced by Travis-CI and PR would be considered broken until checks are passed successfully.
 * Adhere to typical commit guidelines:
-
-    * Commits should not cause NailGun to break.
+    * Commits should not cause NailGun’s unit test to fail. If it does, it will the responsibility of contributor to review those failures and fix them in the same PR's or raise another. The tracking of failures would be responsibility of contributor.
     * Commits should be small and coherent. One commit should address one issue.
     * Commits should have `good commit messages`_.
     * `Rebasing`_ is encouraged. Rebasing produces a much nicer commit history
       than merging.
+* To make the review process easy for all reviewers and anyone else interested in the new functionality please provide some output making use of your changes, this is to be added as a docstring as a example usage for the code added,  this could be in the form of a python interactive shell, test results from Robottelo or Foreman Ansible modules or soemthing else! However you intend to make use of your change will be just fine!
+* When raising a PR against any other branch which is also applicable for '6.X' or foreman branch. Please specify that information in the PR message
+* Unit tests
+    * Unit tests are compulsory
+    * Unit tests should cover all available actions, for the entity. For eg: Repository Sets, have enable, disable, list_available there should be unit tests exercising these actions.
+* When in doubt, ask on IRC. Join #robottelo on `freenode`_
+
+**Important to Note :**
+
+* Define Foreman Version labels in Nailgun
+    The moment PR would be raised, contributor needs to set the right version.
+
+* All PRs should be raised along with Unit tests
+    The unit tests should be added while adding a new entity or modifying the existing entity or modifying and adding to the core of Nailgun.
+
+* Test results from upstream devel or from upstream nightly
+    The API call results are required to make the review process more firm. The PR author can provide results from any library that uses the contributed code by running the changes on upstream nightly or from his/her devel box. The interactive python shell output would be acceptable
+
+Nailgun Review Process
+______________________
+
+* Travis CI is run, and any issues are resolved by contributor.
+* If deemed necessary by contributors/reviewers, an automation run is triggered.
+* At least two ACKs are required to merge a pull request.
+* At least one ACK must be from a Tier 2 reviewer.
+* If a PR requires changes to the CI environment, the “CI Owner” must also provide an ACK.
```

Comment:
What do we mean by "CI environment" here? Travis-CI for nailgun or upstream/downstream Jenkins?

If Jenkins, does it even applies here? As far as I know, Foreman/Katello Jenkins owners don't have write permissions here (so they can't ACK), and they aren't particularly interested in what we do.

I might be missing or misunderstanding something, but I feel this point can be safely removed.

Reply:
We don't have a CI runs of test cases yet, this would be applicable once we do, but I see no harm in adding it now

Reply:
I wasn't aware of these plans. Thanks for clarification!